testplan/testing/multitest/base.py

```diff
@@ -1158,6 +1158,9 @@ def _run_testcase(
                 ),
             )
 
+        if getattr(self.cfg, "collect_code_context", False):
+            testcase = collect_code_context(func=testcase)
```

Comment:
This doesn't make sense, the decorator sets a threading.local variable, why are do calling it on every testcase?

Reply:
I have refactored the solution and added using Result for storing the flag as we discussed it online.